File: examples/identity/create_credential/create_credential.go
package main

import (
	"log"
	"os"
	"time"

	"github.com/bloock/bloock-sdk-go/v2"
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/identity"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

func main() {
	// we set the API key and create a client
	bloock.ApiKey = os.Getenv("API_KEY")
	// we set de identity managed API host you have deployed
	bloock.IdentityApiHost = os.Getenv("IDENTITY_MANAGED_API_HOST")

	// initialize the Key Client
	keyClient := client.NewKeyClient()
	// initialize the IdentityClient
	identityClient := client.NewIdentityClient()

	// we must have our Issuer Baby JubJub key identifier. Ex: 6f36448d-49f3-4b0e-aa72-6e55863302e8
	savedIssuerKey := "6f36448d-49f3-4b0e-aa72-6e55863302e8"
	loadedManagedKey, err := keyClient.LoadManagedKey(savedIssuerKey)
	if err != nil {
		log.Fatalln(err)
	}

	// if we don't have our Issuer entity, here you can import you Issuer from the key
	importedIssuerKey := key.Key{ManagedKey: &loadedManagedKey}
	issuerMethodDID := identity.PolygonID

	importedIssuer, err := identityClient.ImportIssuer(importedIssuerKey, issuerMethodDID)
	if err != nil {
		log.Fatalln(err)
	}

	// we create the credential, passing credential information
	schemaCID := "QmadTvnNKvj2fBDgen35uAp1TfP9pSPVCNeDWw4fitqqne"                        // schema identifier created before. Ex: QmadTvnNKvj2fBDgen35uAp1TfP9pSPVCNeDWw4fitqqne. REQUIRED.
	holderDID := "did:polygonid:polygon:main:2q544HUegzeRpwr3V2qu9eMwgrAmF5x4E1NCPzbQc4" // the DID of the Holder to whom the credential will be associated. REQUIRED.
	expiration := int64(4089852142)                                                      // unix timestamp when the credential will expire. REQUIRED.
	credentialVersion := int32(0)                                                        // credential version. By default it's set to 0. REQUIRED.

	receipt, err := identityClient.BuildCredential(importedIssuer, schemaCID, holderDID, expiration, credentialVersion).
		WithIntegerAttribute("number", 1).
		WithDecimalAttribute("salary", 3000.70).
		WithStringAttribute("nif", "54688188M").
		WithBooleanAttribute("previous_formation", true).
		WithDateAttribute("birth_date", time.Date(1999, time.March, 20, 0, 0, 0, 0, time.UTC)).
		WithDatetimeAttribute("time_registered", time.Now()).
		WithStringAttribute("country", "spain").
		WithIntegerAttribute("brother", 1).
		WithDecimalAttribute("height", 1.70).
		Build()
	if err != nil {
		log.Fatalln(err)
	}

	// returns a Credential entity
	log.Println(receipt.CredentialId)   // Credential identifier. It's an UUID. Ex: f1cae203-6b54-4d71-a6e4-00e73e0c45a5
	log.Println(receipt.Credential)     // Verifiable Credential, the credential entity.
	log.Println(receipt.CredentialType) // the credential schema type associated.
}

File: examples/identity/create_credential/create_credential.py
import os
import datetime
import bloock
from bloock.client.identity import IdentityClient
from bloock.client.key import KeyClient
from bloock.entity.key.key import Key
from bloock.entity.identity.did_method import DidMethod

# we set the API key and create a client
bloock.api_key = os.environ["API_KEY"]
# we set de identity managed API host you have deployed
bloock.identity_api_host = os.environ["IDENTITY_MANAGED_API_HOST"]

# initialize the Key Client
key_client = KeyClient()
# initialize the IdentityClient
identity_client = IdentityClient()

# we must have our Issuer Baby JubJub key identifier. Ex: 6f36448d-49f3-4b0e-aa72-6e55863302e8
saved_issuer_key = "6f36448d-49f3-4b0e-aa72-6e55863302e8"
loaded_managed_key = key_client.load_managed_key(saved_issuer_key)

# if we don't have our Issuer entity, here you can import you Issuer from the key
imported_issuer_key = Key(loaded_managed_key)
issuer_method_did = DidMethod.PolygonID

imported_issuer = identity_client.import_issuer(
    imported_issuer_key, issuer_method_did)

# we create the credential, passing credential information

# schema identifier created before. Ex: QmadTvnNKvj2fBDgen35uAp1TfP9pSPVCNeDWw4fitqqne. REQUIRED.
schema_cid = "QmadTvnNKvj2fBDgen35uAp1TfP9pSPVCNeDWw4fitqqne"
# the DID of the Holder to whom the credential will be associated. REQUIRED.
holder_did = "did:polygonid:polygon:main:2q544HUegzeRpwr3V2qu9eMwgrAmF5x4E1NCPzbQc4"
# unix timestamp when the credential will expire. REQUIRED.
expiration = 4089852142
# credential version. By default it's set to 0. REQUIRED.
credential_version = 0

receipt = identity_client.build_credential(imported_issuer, schema_cid, holder_did, expiration, credential_version) \
    .with_integer_attribute("number", 1) \
    .with_decimal_attribute("salary", 3000.70) \
    .with_string_attribute("nif", "54688188M") \
    .with_boolean_attribute("previous_formation", True) \
    .with_date_attribute("birth_date", datetime.date(1999, 3, 20)) \
    .with_datetime_attribute("time_registered", datetime.datetime.now()) \
    .with_string_attribute("country", "spain") \
    .with_integer_attribute("brother", 1) \
    .with_decimal_attribute("height", 1.70) \
    .build()

# returns a Credential entity

# Credential identifier. It's an UUID. Ex: f1cae203-6b54-4d71-a6e4-00e73e0c45a5
print(receipt.credential_id)
# Verifiable Credential, the credential entity.
print(receipt.credential)
# the credential schema type associated.
print(receipt.credential_type)